app/assets/javascripts/components/story/ExpandedStory/ExpandedStoryControls.js

```diff
@@ -1,33 +1,52 @@
-import React from 'react';
-
-const ExpandedStoryControls = ({ onCancel, onSave, onDelete, readOnly }) => {
-  return (
-    <div className="form-group Story__controls">
-      <input className="save"
-        onClick={onSave}
-        type="button"
-        value={I18n.t('save')}
-        disabled={readOnly}
-      />
-
-      <input className="delete"
-        onClick={() => {
-          if (window.confirm(I18n.t('story destroy confirm'))){
-            onDelete();
-          }
-        }}
-        type="button"
-        value={I18n.t('delete')}
-        disabled={readOnly}
-      />
-
-      <input className="cancel"
-        onClick={onCancel}
-        type="button"
-        value={I18n.t('cancel')}
-      />
-    </div>
-  );
+import React, { Component } from 'react';
+
+class ExpandedStoryControls extends Component {
+  constructor(props) {
+    super(props);
+
+    this.handleCancel = this.handleCancel.bind(this);
+  }
+
+  handleCancel() {
+    const { onCancel, isDirty } = this.props;
+
+    if (isDirty && !window.confirm(I18n.t('story unsaved changes'))) {
+      return;
+    }
+    onCancel();
```

Comment:
I think you could write just like that:

```
if (isDirty && hasNotUnsavedChanges) {
  onCancel();
}
```
Right?

Reply:
Without the context of it, I'd prefer`isDirty && savedAllChanges` since `hasNotUnsavedChanges` has "two negatives" and can be confusing.

Reply:
Yeah, totally right. But I'm not sure if he's talking about `savedAllChanges`. Definitely, there's a better option.

Reply:
Yeah, I can see how that can look a little bit confusing, I'm working on it, thanks!